NB I tried using this new clang version to build Pigweed. Mostly it works, but unfortunately I ran into many building warnings (some treated as errors) in direct (boringssl) and indirect (zlib, via rules_proto_grpc; old absl, via rules_fuzzing) dependencies. My least favorite is the large number of warnings like,

```
external/zlib/compress.c:22:13: warning: a function definition without a prototype is deprecated in all versions of C and is not supported in C2x [-Wdeprecated-non-prototype]          
int ZEXPORT compress2 (dest, destLen, source, sourceLen, level) 
```

that appear in zlib 1.2 (since they suggest we won't be able to build with `-std=c++20`, and simply silencing them is not an option). We're unfortunately stuck with `rules_proto_grpc` for now (https://issues.pigweed.dev/issues/234873954).

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -32,7 +32,6 @@ jobs:
       - name: Fetch external test deps
         run: |
           bazelisk --bazelrc=ci.bazelrc fetch @com_google_googletest//... \
-            @com_github_google_benchmark//... \
             @com_google_absl//...
 
   build_and_test:
@@ -43,7 +42,6 @@ jobs:
       matrix:
         lib:
           - "@com_google_googletest//..."
-          - "@com_github_google_benchmark//..."
           - "@com_google_absl//..."
 
     # Steps represent a sequence of tasks that will be executed as part of the job

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -15,29 +15,9 @@ on:
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
 
-  fetch_test_deps:
-    runs-on: ubuntu-20.04
-
-    # Steps represent a sequence of tasks that will be executed as part of the job
-    steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v3
-
-      - name: Mount bazel repository cache
-        uses: actions/cache@v3
-        with:
-          path: "~/bazel/repository_cache"
-          key: ${{ runner.os }}-bazel
-
-      - name: Fetch external test deps
-        run: |
-          bazelisk --bazelrc=ci.bazelrc fetch @com_google_googletest//... \
-            @com_google_absl//...
-
   build_and_test:
     # The type of runner that the job will run on
     runs-on: ubuntu-20.04
-    needs: fetch_test_deps
     strategy:
       matrix:
         lib:

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -22,7 +22,9 @@ jobs:
       matrix:
         lib:
           - "@com_google_googletest//..."
-          - "@com_google_absl//..."
+          # Some tests are excluded because of
+          # https://github.com/abseil/abseil-cpp/issues/1378
+          - "-- @com_google_absl//... -@com_google_absl//absl/time:time_test -@com_google_absl//absl/time/internal/cctz:time_zone_format_test -@com_google_absl//absl/time/internal/cctz:time_zone_lookup_test -@com_google_absl//absl/time:time_benchmark"
 
     # Steps represent a sequence of tasks that will be executed as part of the job
     steps:

--- a/WORKSPACE
+++ b/WORKSPACE
@@ -31,6 +31,7 @@ git_repository(
     name = "io_bazel_stardoc",
     commit = "8f6d22452d088b49b13ba2c224af69ccc8ccbc90",
     remote = "https://github.com/bazelbuild/stardoc.git",
+    shallow_since = "1620849756 -0400",
 )
 
 # Sets up Bazels packaging rules, for use the document generator.

--- a/rules_cc_toolchain_deps.bzl
+++ b/rules_cc_toolchain_deps.bzl
@@ -39,6 +39,7 @@ def rules_cc_toolchain_deps():
             name = "rules_os",
             commit = "68cdf228f8449a2b42b3a7b6d65395af74a007d7",
             remote = "https://github.com/silvergasp/rules_os.git",
+            shallow_since = "1624010567 +0800",
         )
 
     # Setup x64 linux sysroot

